### PR TITLE
Fix/3174 Exposure manager call all completions

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR fixes the completion callbacks of exposure detection.
When exposure detection is currently running, call completion with an error.

## Link to Jira
[3174](https://jira.itc.sap.com/browse/EXPOSUREAPP-3174)

### CODE REVIEW

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate.swift [hunk 1]
```diff
@@ -40,6 +40,7 @@ extension AppDelegate: ExposureSummaryProvider {
 		activityStateDelegate: ActivityStateProviderDelegate? = nil,
 		completion: @escaping (ENExposureDetectionSummary?) -> Void
 	) -> CancellationToken {
+		Log.info("AppDelegate: Detect exposure.", log: .riskDetection)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/AppDelegate.swift:43]: <3

File: src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift [hunk 2]
```diff
@@ -29,6 +29,10 @@ enum ExposureNotificationError: Error {
 	case unknown(String)
 }
 
+enum ExposureDetectionError: Error {
+	case exposureDetectionIsRunning
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift:33]:
```suggestion
	case isAlreadyRunning
```

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift [hunk 3]
```diff
@@ -11,6 +11,8 @@ extension OSLog {
     static let ui = OSLog(subsystem: subsystem, category: "ui")
     /// Local data & caches
     static let localData = OSLog(subsystem: subsystem, category: "localdata")
+	/// Risk Detection
+	static let riskDetection = OSLog(subsystem: subsystem, category: "riskdetection")
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Workers/Logging.swift:15]: Indentation seems a little off

**Review by marcussc — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift]: 👍 renamed!

File: src/xcode/ENA/ENA/Source/AppDelegate.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/AppDelegate.swift:43]: ❤️

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:15]: Looks good on my side 🤷‍♂️

Bildschirmfoto 2020-10-19 um 09 29 11

**Review by nickguendling — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift (hunk 3 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:15]: oh okay :D

**Review by marcussc — APPROVED** [review]

**Pull request merged**